src/host/test/ledger.cpp

```diff
@@ -1036,7 +1026,201 @@ TEST_CASE("Delete committed file from main directory")
+  SnapshotManager snapshots(snapshot_dir, ledger);
+
+  size_t snapshot_interval = 5;
+  size_t snapshot_count = 5;
+
+  INFO("Generate snapshots");
+  {
+    for (size_t i = 1; i < snapshot_interval * snapshot_count;
+         i += snapshot_interval)
+    {
+      // Note: Evidence is assumed to be at snapshot idx + 1
+      snapshots.write_snapshot(
+        i, i + 1, dummy_snapshot.data(), dummy_snapshot.size());
+    }
+
+    REQUIRE_FALSE(snapshots.find_latest_committed_snapshot().has_value());
+  }
+
+  INFO("Commit snapshots");
+  {
+    for (size_t i = 1; i < snapshot_interval * snapshot_count;
+         i += snapshot_interval)
+    {
+      // Note: Evidence is assumed to be at snapshot idx + 1
+      snapshots.commit_snapshot(i, dummy_receipt.data(), dummy_receipt.size());
+
+      auto latest_committed_snapshot =
+        snapshots.find_latest_committed_snapshot();
+      REQUIRE(latest_committed_snapshot.has_value());
+      const auto& snapshot = latest_committed_snapshot.value();
+      REQUIRE(get_snapshot_idx_from_file_name(snapshot) == i);
+      REQUIRE(get_snapshot_evidence_idx_from_file_name(snapshot) == i + 1);
+      REQUIRE_FALSE(
+        get_evidence_commit_idx_from_file_name(snapshot).has_value());
+    }
+  }
+}
+
+std::string generate_1_x_snapshot(
+  const std::string& snapshot_dir, size_t snapshot_idx, bool committed)
+{
+  auto snapshot_file_name = fmt::format(
+    "{}{}{}{}{}",
+    snapshot_file_prefix,
+    snapshot_idx_delimiter,
+    snapshot_idx,
+    snapshot_idx_delimiter,
+    snapshot_idx + 1);
+
+  if (committed)
+  {
+    snapshot_file_name =
+      fmt::format("{}.committed_{}", snapshot_file_name, snapshot_idx + 2);
+  }
+  auto full_snapshot_path =
+    fs::path(snapshot_dir) / fs::path(snapshot_file_name);
+
+  std::ofstream snapshot_file(
+    full_snapshot_path, std::ios::out | std::ios::binary);
+  snapshot_file.write(
+    reinterpret_cast<const char*>(dummy_snapshot.data()),
+    dummy_snapshot.size());
+
+  return snapshot_file_name;
+}
+
+std::optional<std::string> commit_1_x_snapshot(
+  const std::string& snapshot_dir, size_t snapshot_idx)
+{
+  for (auto const& f : fs::directory_iterator(snapshot_dir))
+  {
+    auto file_name = f.path().filename().string();
+    if (
+      !is_snapshot_file_committed(file_name) &&
+      get_snapshot_idx_from_file_name(file_name) == snapshot_idx)
+    {
+      const auto committed_file_name = fmt::format(
+        "{}.{}_{}", file_name, snapshot_committed_suffix, snapshot_idx + 2);
+
+      fs::rename(
+        fs::path(snapshot_dir) / fs::path(file_name),
+        fs::path(snapshot_dir) / fs::path(committed_file_name));
+
+      return committed_file_name;
+    }
+  }
+  return std::nullopt;
+}
+
+TEST_CASE(
+  "Backwards compatibility with 1.x snapshots" *
+  doctest::test_suite("snapshot"))
+{
+  // To be removed as part of https://github.com/microsoft/CCF/issues/2981
+  fs::remove_all(ledger_dir);
+  fs::remove_all(snapshot_dir);
```

Comment:
This pattern of removing old dirs at the start of tests seems fragile. If we forget to do it in a new test case, it may work until the tests run in the wrong order. Can we instead ensure these are deleted with a RAII wrapper (ie `class FolderDeleter{ FolderDeleter(name); ~FolderDeleter() { fs::remove_all(name); } };`, `auto ledger_dir_deleter = std::make_shared<FolderDeleter>(ledger_dir);`)? If we want to keep the files around for manual inspection after a failed test, the local modification is just to not create that deleter.